### DISCUSSION TITLE
Why all fields are re rendering if I unregistered only one field?

### DISCUSSION THREAD

**Discussion by sekarcrazy** [react-hook-form/react-hook-form#5290] [Q&A] [answered]
I have almost 40 custom fields in the form. Out of 40, very few fields will be loaded/unloaded on demand based on one field(team).
Now the problem is, When I unregister during component unmount for disappeared field, rest 39 fields are getting re rendered. How to prevent this?
If I am not doing unregister, Form still holds all the values of disappeared fields and validation rule are being applied for thw same.

Any suggestion?

In the sample, I have taken only two inputs for testing purpose.

https://codesandbox.io/s/react-hook-form-unregister-v6-forked-pznw4

**Comment by leapful** [comment, marked as answer]
@sekarcrazy 

It seems you are mis-understanding the concept of FormContext. In your CSB sample, "useFormContext" is used inside the "Input" component, therefore, it will be always rendered following the change of FormContext.

Furthermore, "useMemo" hook is not used to avoid re-rendering component, its purpose is for memoize expensive computed value. React.memo should be used instead of "useMemo".

Working sample: https://codesandbox.io/s/react-hook-form-unregister-v6-forked-1eu2d?file=/src/index.js

**Reply by bluebill1049** [reply]
thank you @leapful

**Reply by sekarcrazy** [reply]
Thanks for your suggestion @leapful . The sample that you provided works with "register" But, If I use custom control using "Controller", Input elements are re rendering by un registering only one field from the form.

Is there a way that I can prevent ! Kindly check this one.

Your response is most appreciated.

https://codesandbox.io/s/react-hook-form-unregister-v6-forked-1j60b

**Reply by leapful** [reply]
@sekarcrazy 

"Controller" component is using pattern "render" prop, it will call "render" prop if there is any change from the form.

In your CSB sample, you are always set a new object for Input component props, therefore, it's always re-rendering even there is no change.

```

<Controller render={({ field: { value, onChange } }) => <Input {...{ value, onChange }} />} />

```

Working sample: https://codesandbox.io/s/react-hook-form-unregister-v6-forked-3qu02?file=/src/index.js
